docs/architecture/adr-076-sign-mode-unified.md

```diff
@@ -0,0 +1,327 @@
+
+* The implementation will include all the necessary code to build and support the new sign mode, including:
+    * Implementation of the MetadataDigest structure and its associated hash calculation function.
+    * Development of the Merkle tree construction and verification logic for the metadata.
+    * Integration of the new sign mode into the existing transaction signing and verification pipeline.
+* A tool will be created to build the metadata file from the existing protobuf files. This tool will:
+    * Parse all relevant protobuf files in the Cosmos SDK.
+    * Generate a structured JSON metadata file that includes all type and message definitions.
+    * A CI job will be implemented to automatically run this metadata generation tool.
+
+* The transaction signing process will be updated to include the MetadataDigest hash in the sign doc.
+* The Ledger HW signing process will be accommodated to match this new sign mode.
+* The transaction verification process will be modified to validate the metadata hash against the locally stored metadata on validator nodes.
+* Comprehensive documentation for `SIGN_MODE_UNIFIED` will be created, including guides for wallet developers and chain developers on how to implement and use the new sign mode.
+* A migration strategy will be implemented to allow for a (progressive) smooth transition from existing sign modes to `SIGN_MODE_UNIFIED`.
+* Finally, all legacy sign modes will be deprecated and removed from the SDK.
+
+## Consequences
+
+### Backwards Compatibility
+
+**Preservation of Existing Sign Modes:**
+* Existing sign modes remain supported to ensure that legacy wallets and clients continue to function without interruption.
+
+### Positive
+
+* **Enhanced User Experience:**
+    * `SIGN_MODE_UNIFIED` provides a more intuitive and secure signing experience for users, especially on hardware wallets, by enabling the display of human-readable transaction details.
+
+* **Improved Security:**
+    * Reduces the risk of blind signing by allowing users to verify transaction contents before signing.
+
+* **Consistency Across Platforms:**
+    * Standardizes the signing process across different wallets and platforms, ensuring uniform behavior and reducing inconsistencies.
+
+* **Reduced Maintenance Overhead:**
+    * Consolidates multiple sign modes into a unified approach, decreasing the complexity and maintenance efforts on the codebase.
+
+* **Versioned Metadata:**
+    * The metadata will be versioned, allowing for better control and management of changes over time.
+
+### Negative
+
+* **Development Effort:**
+    * Requires significant development effort to implement the new sign mode, including updates to multiple components such as transaction builders, signers, and validators and HW wallet apps.
+
+* **Tooling and Documentation Updates:**
+    * Requires updates to existing tools, documentation, and developer guides to accommodate the new sign mode, which may involve additional resources and time.
```

Comment:
Not a show stopper but I think it's worth mentioning that the sign doc is not human readable anymore

Reply:
Added using plain json + spec as alternative to cbor to keep human readability